src/OrleansServiceFabricUtils/OrleansFabricSilo.cs

```diff
@@ -0,0 +1,193 @@
+﻿namespace Microsoft.Orleans.ServiceFabric
+{
+    using System;
+    using System.Diagnostics;
+    using System.Fabric;
+    using System.Net;
+    using System.Threading.Tasks;
+
+    using global::Orleans.Runtime;
+
+    using global::Orleans.Runtime.Configuration;
+
+    using global::Orleans.Runtime.Host;
+
+    /// <summary>
+    /// Wrapper class for an Orleans silo running in the current host process.
+    /// </summary>
+    internal class OrleansFabricSilo
+    {
+        /// <summary>
+        /// The task which completes when the silo stops running.
+        /// </summary>
+        private readonly TaskCompletionSource<int> stopped = new TaskCompletionSource<int>();
+
+        /// <summary>
+        /// The silo name.
+        /// </summary>
+        private readonly string siloName;
+
+        /// <summary>
+        /// The service name, sanitized for use as a deployment id.
+        /// </summary>
+        private readonly string escapedServiceName;
+
+        /// <summary>
+        /// The host.
+        /// </summary>
+        private SiloHost host;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OrleansFabricSilo"/> class.
+        /// </summary>
+        /// <param name="context">The service context.</param>
+        /// <param name="siloEndpoint">
+        /// The silo endpoint.
+        /// </param>
+        /// <param name="proxyEndpoint">
+        /// The proxy endpoint.
+        /// </param>
+        public OrleansFabricSilo(
+            ServiceContext context,
+            IPEndPoint siloEndpoint, 
+            IPEndPoint proxyEndpoint)
+        {
+            this.SiloEndpoint = siloEndpoint;
+            this.ProxyEndpoint = proxyEndpoint;
+            this.escapedServiceName = OrleansFabricUtility.EscapeServiceName(context.ServiceName);
+            this.siloName = this.escapedServiceName + "_" + context.ReplicaOrInstanceId.ToString("X");
+        }
+
+        /// <summary>
+        /// Gets the silo endpoint.
+        /// </summary>
+        public IPEndPoint SiloEndpoint { get; }
+
+        /// <summary>
+        /// Gets the proxy endpoint.
+        /// </summary>
+        public IPEndPoint ProxyEndpoint { get; }
+
+        /// <summary>
+        /// Gets the task which completes when the silo stops running.
+        /// </summary>
+        public Task Stopped => this.stopped.Task;
+
+        /// <summary>
+        /// Gets the unique address of this silo.
+        /// </summary>
+        public SiloAddress Address => SiloAddress.New(this.host.NodeConfig.Endpoint, this.host.NodeConfig.Generation);
+
+        /// <summary>
+        /// Starts the silo.
+        /// </summary>
+        /// <param name="config">
+        /// The config.
+        /// </param>
+        /// <exception cref="OrleansException">
```

Comment:
Fixed. I've removed that from the docs since I'm no longer explicitly throwing an exception in that method.